### DISCUSSION TITLE
MyScaleDB now supports Full-Text and Hybrid Search

### DISCUSSION THREAD

**Discussion by usamajamil43** [community/community#122300] [Other Feature Feedback, Questions, & Ideas]
In version 1.5.0 of MyScale, we introduced an upgraded full-text search feature powered by [Tantivy](https://github.com/quickwit-oss/tantivy). Tantivy has a lower latency rate and it's written in Rust.

In the latest update, MyScaleDB now supports:

- Supports full-text search
- Supports fuzzy and wildcard searches along with rich tokenizers
- Utilizes BM25 for relevance scoring similar to Elasticsearch
- Query times over 5M rows are 300x faster than ClickHouse's built-in inverted index
- Real-time searching without manual reindexing

For a more detailed explanation, you can read this blog: [Introducing MyScale's Powerful Full-Text and Hybrid Search Capabilities](https://myscale.com/blog/text-search-and-hybrid-search-in-myscale/)

or take a look at these documents:
- [Full-Text Search in MyScale](https://myscale.com/docs/en/text-search/)
- [Hybrid Search in MyScale](https://myscale.com/docs/en/hybrid-search/)

**Comment by LiteBrite82** [comment]
Hi @usamajamil43 , we’re glad you’re here 😀

Thanks for sharing your program update with the community. Unfortunately, we currently do not allow self-promotion or advertising on the Community Discussions. We want to make sure there is space for users to ask questions without overwhelming them with other conversations. Thank you for helping us maintain a productive and tidy community for all our members.
